### PR TITLE
Add tf.data.Dataset support for orca tf estimator 

### PR DESCRIPTION
**Pull request by jenniew**
Add tf.data.Dataset support for orca tf estimator train, eval, predict.
Predict result is xshards.

### CODE REVIEW

File: pyzoo/zoo/orca/learn/tf/utils.py [hunk 1]
```diff
@@ -125,6 +137,21 @@ def combine(pair):
     return result_df
 
 
+def convert_predict_to_xshard(prediction_rdd):
+    def transform_predict(iter):
+        predictions = list(iter)
+        # list of np array
+        if isinstance(predictions[0], list):
+            predictions = np.array(predictions).T.tolist()
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf/utils.py:145]: why transpose?

File: pyzoo/zoo/orca/learn/tf/estimator.py [hunk 2]
```diff
@@ -295,6 +297,8 @@ def predict(self, data, batch_size=4,
         predicted_rdd = self.model.predict(dataset, batch_size)
         if isinstance(data, DataFrame):
             return convert_predict_to_dataframe(data, predicted_rdd)
+        elif isinstance(data, tf.data.Dataset):
+            return convert_predict_to_xshard(predicted_rdd)
         else:
             return predicted_rdd
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf/estimator.py:303]: You should document the expected input and output of the predict function

**Comment by jason-dai** [comment]
change the backend from "spark" to "bigdl"

File: pyzoo/zoo/orca/learn/tf/utils.py (hunk 1 above)
Comment (jenniew) [reply pyzoo/zoo/orca/learn/tf/utils.py:145]: if the predict record is a list of numpy arrays, 

> change the backend from "spark" to "bigdl"

Do you mean change OrcaContext.pandas_read_backend value from "spark" to "bigdl"?

**Comment by jenniew** [comment]
> change the backend from "spark" to "bigdl"

Do you mean change OrcaContext.pandas_read_backend from "spark" to "bigdl"

File: pyzoo/zoo/orca/learn/tf/utils.py (hunk 1 above)
Comment (jenniew) [reply pyzoo/zoo/orca/learn/tf/utils.py:145]: This is for the case that prediction result is a list of np arrays. For example, the first element is [np11, np21, np31], the second element is [np12, np22, np32], in mapPartitions, we combine the result and get predictions=[[np11, np21, np31], [np12, np22, np32]], use transpose to change to [[np11, np12], [np21, np22], [np31, np32]] which means the 3 prediction outputs correspondingly.

**Comment by jason-dai** [comment]
> > change the backend from "spark" to "bigdl"
> 
> Do you mean change OrcaContext.pandas_read_backend from "spark" to "bigdl"

Change the backend parameter in tf.Estimator from "spark" to "bigdl"

File: pyzoo/zoo/orca/learn/tf/estimator.py (hunk 2 above)
Comment (jenniew) [reply pyzoo/zoo/orca/learn/tf/estimator.py:426]: add documents.

**Comment by jenniew** [comment]
@jason-dai , can you also review https://github.com/intel-analytics/analytics-zoo/pull/2590? Thanks.

**Comment by jenniew** [comment]
Jenkins pass: http://10.239.47.210:18888/job/ZOO-PR-Validation/3986/

**Pull request merged**